gateway/join.go

```diff
@@ -0,0 +1,254 @@
+package gateway
+
+/*
+#cgo CFLAGS: -I./sx1302/libloragw/inc -I./sx1302/libtools/inc
+#cgo LDFLAGS: -L./sx1302/libloragw -lloragw -L./sx1302/libtools -lbase64 -lparson -ltinymt32  -lm
+
+#include "../sx1302/libloragw/inc/loragw_hal.h"
+#include "gateway.h"
+#include <stdlib.h>
+
+*/
+import "C"
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/crypto"
+	"go.thethings.network/lorawan-stack/v3/pkg/crypto/cryptoservices"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/v3/pkg/types"
+	"go.viam.com/utils"
+)
+
+type JoinRequest struct {
+	joinEUI  []byte
+	devEUI   []byte
+	devNonce []byte
+	mic      []byte
+}
+
+const (
+	joinRx2WindowSec = 6         // rx2 delay for sending join accept message.
+	rx2Frequenecy    = 923300000 // Frequuency for rx2 window
+	rx2SF            = 12        // spreading factor for rx2 window
+	rx2Bandwidth     = 0x06      // 500k bandwidth
+)
+
+var errNoDevice = errors.New("received join request from unknown device")
+
+// network id for the device to identify the network.
+var netID = []byte{1, 2, 3}
```

Comment:
why 1,2,3? Is it significant for the order? If we need a byte array that's just empty by 3 elements 3, could we do var netID [3]byte?

Reply:
its just a random number I put, we just need a 3 byte ID to send the devices that's used to identify our network.

Reply:
So i could be a [3]byte instantiated in go, but it is totally okay to keep it as is with a callout that it has to be a length 3 array, and that the number values don't mean anything for instantiation.